### PR TITLE
Double finger zoom should not select things

### PR DESCRIPTION
**Pull request by vnsteven**
Implements this feature [#1328](https://github.com/excalidraw/excalidraw/issues/1328)

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -2456,9 +2457,15 @@ export class App extends React.Component<any, AppState> {
         delta = MAX_STEP;
       }
       delta *= sign;
+      const newElements = { ...selectedElementIds };
+      Object.keys(newElements).forEach((key) => {
+        newElements[key] = false;
+      });
       this.setState(({ zoom }) => ({
         zoom: getNormalizedZoom(zoom - delta / 100),
+        selectedElementIds: newElements,
```
Comment (vjeux) [inline src/components/App.tsx:2466]: Shouldn’t you just do selectedElementIds: {} ?

Comment (vnsteven) [reply src/components/App.tsx]: You're right, I changed it !

**Comment by j-f1** [comment]
It should also not unselect things. Maybe when the first finger is placed, the selected elements could be recorded then restored when the second finger is placed?

**Comment by vnsteven** [comment]
@j-f1 Since there is no gesture events for non-safari browsers, I wasn't able to restore the selected elements with finger placements. And it wouldn't work for every user. So I did it on the wheel event. What do you think?

**Comment by j-f1** [comment]
Now it doesn’t seem to be working at all (on iPhone) :(

**Comment by vjeux** [comment]
The first version wasn’t working for me either on iphone fwiw

**Comment by vnsteven** [comment]
I handled touch and gesture events so it should works now

**Comment by vnsteven** [comment]
I handled touch and gesture events so it should work now

**Comment by j-f1** [comment]
Not :100:% perfect but definitely much better than the current behavior.

**Comment by vjeux** [comment]
I’ve tried the latest version and here what I see happen:

When you put the second finger it keeps the element selected during the zoom. But when you end the zoom then it unselects the element.

This looks very odd to me. Would it be better to unselect right when the second finger touches or when zoom starts?

**Comment by vnsteven** [comment]
@vjeux It was due to touching an element and zooming at the same time. I handled it by recording the elements when an element is hit and unselect it when zoom starts as you said.

**Comment by lipis** [comment]
@vnsteven could you fix the conflicts and we could get that in?

**Comment by vnsteven** [comment]
@lipis done!

**Comment by lipis** [comment]
Something is wrong with the robots.. Vercel?

**Pull request merged**